### PR TITLE
Headless support

### PR DESCRIPTION
**Pull request by jorisfritzsche**
Add support for headless e-commerce by adding API controllers that mimic current controller functionality.

### CODE REVIEW

File: app/controllers/concerns/spree/adyen/can_confirm_payment.rb [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      def restore_order
+        payment_method_id = params.fetch(:merchantReturnData).split('|').last
+
+        @payment_method = Spree::PaymentMethod.find(payment_method_id)
+        @order = Spree::Order.find_by!(number: order_number)
+      end
+
+      def check_signature
+        unless ::Adyen::HPP::Signature.verify(
+            response_params,
```
Comment (Sytzeandreae) [inline app/controllers/concerns/spree/adyen/can_confirm_payment.rb]: Why not just use the `adyen_permitted_params` here?

File: app/models/spree/adyen/confirm_hpp_payment.rb [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+      payment.source.update(source_params)
+
+      true
+    end
+
+    def confirm_order_incomplete
+      source = Spree::Adyen::HppSource.new(@source_params)
+
+      return false unless source.authorised?
```
Comment (Sytzeandreae) [inline app/models/spree/adyen/confirm_hpp_payment.rb]: Why return a boolean here?

File: app/controllers/concerns/spree/adyen/can_confirm_payment.rb (hunk 1 above)
Comment (jorisfritzsche) [inline app/controllers/concerns/spree/adyen/can_confirm_payment.rb]: Excellent question. Will update

File: app/models/spree/adyen/confirm_hpp_payment.rb (hunk 2 above)
Comment (Sytzeandreae) [inline app/models/spree/adyen/confirm_hpp_payment.rb]: Okay, I didn't see the confirm method being used somewhere, so I wondered...

**Pull request merged**